core/worker.go

```diff
@@ -56,6 +57,10 @@ const (
 
 	// c_chainSideChanSize is the size of the channel listening to uncle events
 	chainSideChanSize = 10
+
+	c_uncleCacheSize = 50
```

Comment:
So no more than 50 workshares can be in a block, correct?

Reply:
only 16 work shares in a block, but this is how many the worker handles in its cache, i can increase this number as well if we like

Reply:
we should probably have better way to manage this work share list, but this is just a basic implementation

Reply:
Increased the size to 100

Reply:
params.MaxWorkShareCount is the actual limit on the workshares in block